server/aws/mc-lambda.tf

```diff
@@ -0,0 +1,106 @@
+##
+#  Mectrics Collection Lambda
+##
+
+resource "aws_lambda_function" "metrics" {
+  function_name = var.service_name
+  description = var.lambda-description
+  s3_bucket = var.lambda_code
+  s3_key = var.lambda-function-code
+  handler = var.lambda-function-handler
+  runtime = var.lambda-function-runtime
+  role = aws_iam_role.role.arn
+  environment {
+    variables = {
+      dataBucket = "${var.s3_raw_metrics_bucket_name}-${data.aws_caller_identity.current.account_id}"
+      dataKey = aws_kms_key.mykey.id
```

Comment:
We don't need to double encrypt data if we are already encrypting the bucket.

Reply:
This isn't double encrypting. This is the KMS key environment variable required by the Lambda policy to allow the Lambda function to encrypt and put objects in the S3 bucket. However since we are changing to AES256 it is not needed and the function key type will be changed. This is to inform the Lambda function what type encryption is being used and the method for accessing the key itself. This line will be removed since we are using an AWS managed now.